soc/mediatek/mt8xxx/soc.c

```diff
@@ -173,7 +173,7 @@ __asm__(".align 4\n\t"
  * a 1024 byte alignment requirement, forcing us to waste ~1011 bytes
  * needlessly.
  */
-__asm__(".pushsection .sof_entry.text\n\t"
+__asm__(".pushsection .sof_entry.text, \"ax\"\n\t"
```

Comment:
Hm... not sure about this?  This section can't be part of any linker defaults, it really needs to be placed precisely at the beginning of the image to conform with the entry point set by the kernel driver.

It's not wrong to set the flags, but if it's not going where you need it the bug is in the linker script.